src/Index.js

```diff
@@ -928,6 +928,9 @@ Index.prototype.findObject = function(findCallback, requestOptions, callback) {
   var paginate = requestOptions.paginate !== undefined ? requestOptions.paginate : true;
   var query = requestOptions.query !== undefined ? requestOptions.query : '';
 
+  delete requestOptions.paginate;
+  delete requestOptions.query;
```

Comment:
doesn't look like query is actually removed here according to the test, is that expected?

Reply:
Without the delete, the test will fail having this result:

```
# findObject: object not found with non matching query
not ok 3 should be equal
  ---
    operator: equal
    expected: |-
      '{"params":"query=Algolia&page=0"}'
    actual: |-
      '{"params":"query=Algolia&query=Algolia&page=0"}'
```

Reply:
This is expected, the query should passed within params, and not as a new key on the payload.

Reply:
makes sense, I misread the variable names